lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -493,10 +489,10 @@ defmodule ExUnit.Assertions do
       end
 
     timeout =
-      if is_integer(timeout) do
-        timeout
-      else
-        quote do: ExUnit.Assertions.__timeout__(unquote(timeout))
+      case timeout do
```

Comment:
Unfortunately the value can be `nil` at runtime but a var at compilation time, so I think we can always call `ExUnit.Assertions.__timeout__` instead and handle `nil` and defaults in there. WDYT?

Reply:
Yeah, that works better.